src/components/PostDetails.tsx

```diff
@@ -1,118 +1,64 @@
-import React from 'react';
-import { Loader } from './Loader';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { Comments } from './Comments';
+import { useComments } from './Contexts/CommentsContext';
 import { NewCommentForm } from './NewCommentForm';
 
-export const PostDetails: React.FC = () => {
-  return (
-    <div className="content" data-cy="PostDetails">
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
-
-          <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
-          </p>
-        </div>
-
-        <div className="block">
-          <Loader />
-
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
-
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
-
-          <p className="title is-4">Comments:</p>
+type Props = {
+  onAddComment: (status: boolean) => void,
+  onDeleteComment: (id: number) => void,
+  comments: Comment[],
+  isNewComment: boolean,
+  hasError: boolean,
+  selectedPost: Post,
+};
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+export const PostDetails: React.FC<Props> = ({
+  onAddComment,
+  onDeleteComment,
+  comments,
+  isNewComment,
+  hasError,
+  selectedPost,
+}) => {
+  const { setComments } = useComments();
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
+  const toggleAddCommentForm = () => {
+    onAddComment(true);
+  };
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a
-                href="mailto:misha@mate.academy"
-                data-cy="CommentAuthor"
-              >
-                Misha Hrynko
-              </a>
+  const handleDeleteComment = (commentId: number) => {
+    setComments(prev => prev.filter(({ id }) => id !== commentId));
+    onDeleteComment(commentId);
+  };
```

Comment:
I didn't find where you catch API errors, if you don't catch an error your app can crash.
Also, filter comments only if the comment was successfully deleted

Reply:
It is written in the task to filter comments before delete

> Implement comment deletion
Delete the commnet immediately not waiting for the server response to improve the UX.

Reply:
> It is written in the task to filter comments before delete
> 
> > Implement comment deletion
> > Delete the commnet immediately not waiting for the server response to improve the UX.

Sure, but you still need to handle possible API errors